### PR TITLE
feat: add authentication tokens to bot service

### PR DESCRIPTION
**Pull request by ghost**
adds a way to get and refresh the authentication token for a bot.

this implies bots do not use the user auth flow and just have a persistent session token - should that be documented anywhere?

### CODE REVIEW

File: staging/bots/v1/bots.proto [hunk 1]
```diff
@@ -65,6 +67,15 @@ message EditBotRequest {
 // Response type for EditBot
 message EditBotResponse {}
 
+// Request type for RefreshToken
+message RefreshTokenRequest {
+  uint64 bot_id = 1;
```
Comment (90-008) [inline staging/bots/v1/bots.proto:74]: Needs doc comment

File: staging/bots/v1/bots.proto [hunk 2]
```diff
@@ -65,6 +67,15 @@ message EditBotRequest {
...
+}
+// Response type for RefreshToken
+message RefreshTokenResponse {
+  string new_token = 1;
```
Comment (90-008) [inline staging/bots/v1/bots.proto:79]: Needs doc comment

**Review by 90-008 — COMMENTED** [review]
> this implies bots do not use the user auth flow and just have a persistent session token - should that be documented anywhere?

This should probably be documented in `auth` service, because that's where people will look for this first.

**Comment by ghost** [comment]
@yusdacra I updated the PR based on your comments

File: stable/auth/v1/auth.proto [hunk 3]
```diff
@@ -289,6 +289,8 @@ message CheckLoggedInResponse {}
 // TODO: finish
 
 // The service containing authorization/entication methods
+//
+// It is important to note that bots are exempt from AuthService - a persistent authorization token is generated on bot creation.
```
Comment (90-008) [inline stable/auth/v1/auth.proto]: It would be good if bots service was mentioned here

**Review by 90-008 — CHANGES_REQUESTED** [review]

File: stable/auth/v1/auth.proto (hunk 3 above)
Comment (ghost) [reply stable/auth/v1/auth.proto]: is it possible to hyperlink it?

**Review by 90-008 — APPROVED** [review]

**Pull request merged**